skimage/restoration/_denoise.py

```diff
@@ -459,8 +466,13 @@ def _wavelet_threshold(img, wavelet, threshold=None, sigma=None, mode='soft',
     if threshold is None:
         # The BayesShrink thresholds from [1]_ in docstring
         var = sigma**2
-        threshold = [{key: _bayes_thresh(level[key], var) for key in level}
-                     for level in dcoeffs]
+        if method == "BayesShrink":
+            # The BayesShrink thresholds from [1]_ in docstring
+            threshold = [{key: _bayes_thresh(level[key], var) for key in level}
+                         for level in dcoeffs]
+        else:
+            threshold = [{key: _universal_thresh(img, level[key], sigma)
```

Comment:
There is no need to call `_universal_thresh` each time as the threshold is the same for all coefficient levels.  I would store the threshold in a variable and then build the dict using that single threshold